python/tskit/trees.py

```diff
@@ -4677,6 +4683,107 @@ def genotype_matrix(
             isolated_as_missing=isolated_as_missing, alleles=alleles
         )
 
+    def alignments(
+        self,
+        *,
+        reference_sequence=None,
+        isolated_as_missing=None,
+        missing_data_character=None,
+    ):
+        """
+        Returns an iterator over the full sequence alignments for the samples in this
+        tree sequence. Each alignment ``a`` is a string of length ``L`` where L
+        is the sequence length of this tree sequence, and ``a[j]`` is the nucleotide
+        value for position ``j`` on the sequence.
+
+        .. note:: This is inherently a **zero-based** representation of the sequence
+            coordinate space. Care will be needed when interacting with other
+            libraries and upstream coordinate spaces.
+
+        If a ``reference_sequence`` is supplied this will be used to fill
+        in the nucleotide positions between the sites in the tree sequence; otherwise,
+        nucleotides between sites are treated as missing data and therefore filled
+        in with the ``missing_data_character`` (by default "-").
+
+        .. warning:: The future behaviour of this function will change when the
```

Comment:
I'm wondering if it might be better here to error out when reference_sequence=None with a message saying that this will be supported in future?